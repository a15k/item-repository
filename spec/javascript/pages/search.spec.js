import renderer from 'react-test-renderer';
import { MemoryRouter as R } from 'react-router';
import Search from 'pages/search';
import AssessmentPreview from 'models/assessment/preview';
import SearchCollection from 'pages/search/collection';
import Factory from '../factories';

describe(Search, () => {
  let props;

  beforeEach(() => {
    const collection = new SearchCollection();
    AssessmentPreview.assets = {
      javascript: '/preview.js', stylesheet: 'preview.css',
    };
    collection.fromJSON([ Factory.create('Assessment') ]);
    props = {
      collection,
    };
  });

  it('renders help', () => {
    props.collection.assessments.clear();
    const search = renderer.create(<R><Search {...props} /></R>);
    expect(search.toJSON()).toMatchSnapshot();
    search.unmount();
  });

  it('renders search', () => {
    const search = renderer.create(<R><Search {...props} /></R>);
    expect(search.toJSON()).toMatchSnapshot();
    search.unmount();
  });

  it('performs search', () => {
    const search = mount(<R><Search {...props} /></R>);
    props.collection.search = jest.fn();
    search.find('Search').instance().inputRef.value = 'one two three';
    search.find('.input-group-append Button').simulate('click');
    expect(props.collection.search).toHaveBeenCalledWith('one two three');
    search.unmount();
  });

});
