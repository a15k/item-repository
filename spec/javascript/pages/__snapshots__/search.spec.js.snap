// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders help 1`] = `
<div
  className="search-page"
>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Search by 
        Text
      </button>
      <div
        aria-hidden={true}
        className="dropdown-menu"
        role="menu"
        tabIndex="-1"
      >
        <button
          className="dropdown-item"
          data-type="ID"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          ID
        </button>
        <button
          className="dropdown-item"
          data-type="Text"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          Text
        </button>
      </div>
    </div>
    <input
      className="form-control"
      onKeyPress={[Function]}
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Enter text to search for.  Assessment metadata is searched first and then statisistics
  </small>
</div>
`;

exports[`Search renders search 1`] = `
.c0 {
  margin-top: 1rem;
  padding-top: 1rem;
  border-top: 1px solid grey;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 > * {
  margin-left: 0.3rem;
}

<div
  className="search-page"
>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Search by 
        Text
      </button>
      <div
        aria-hidden={true}
        className="dropdown-menu"
        role="menu"
        tabIndex="-1"
      >
        <button
          className="dropdown-item"
          data-type="ID"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          ID
        </button>
        <button
          className="dropdown-item"
          data-type="Text"
          onClick={[Function]}
          tabIndex="0"
          type="button"
        >
          Text
        </button>
      </div>
    </div>
    <input
      className="form-control"
      onKeyPress={[Function]}
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="c0"
  >
    <div
      className="c1"
    />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Ipsa qui unde nihil laudantium nihil autem eius voluptas. Vel dignissimos facilis voluptates quae inventore asperiores sit est quae. Cum sint mollitia aut provident qui ipsa voluptatem natus et. Voluptas quas animi aut vitae. Nihil error neque consequatur aperiam et voluptas. Rem temporibus et asperiores eum a sed autem eveniet quis.</p>",
        }
      }
    />
  </div>
</div>
`;
